projects/RabbitMQ.Client/client/api/IBasicPublishBatch.cs

```diff
@@ -37,12 +37,14 @@
 //  The Initial Developer of the Original Code is Pivotal Software, Inc.
 //  Copyright (c) 2007-2020 VMware, Inc.  All rights reserved.
 //---------------------------------------------------------------------------
+
 using System;
 
 namespace RabbitMQ.Client
 {
     public interface IBasicPublishBatch
     {
+        [Obsolete("Use Add(string exchange, string routingKey, bool mandatory, IBasicProperties properties, ReadOnlyMemory<byte> body) instead. Will be replaced in version 7.0", false)]
```

Comment:
Since this is an interface change, we can only ship it in 7.0. Does the last sentence here therefore make sense? Should it be removed?

Reply:
The interface is not changed. See my comment

> It seems that `BasicPublishBatch` doesn't support yet `ReadOnlyMemory<byte>`. This is a proposal to add it in a non-breaking way, which could be added in `6.2`. Later the extension method can become the interface method in v7 because there the breaking change can be made.

the last flag on the Obsolete attribute makes this a warning indicating there is a no way of calling basic publish available. When users use the extension method and transition over there in v7 the interface can be broken and the obsolete removed

// cc @bording

Reply:
Sorry, I've misread what override the attribute is attached to.